src/utils/windowUtils.ts

```diff
@@ -41,6 +41,35 @@ export module Utils {
       }
     }
 
+    /**
+     * Sends a warning to the console. The warning includes the version number of the deprecation,
+     * the name of the function which was deprecated and an optional message.
+     *
+     * Tu be used in the first line of a deprecated method.
+     *
+     * @param {string} version The version when the tagged method became obsolete
+     * @param {string?} message Optional message to be shown with the warning
+     */
+    export function deprecated(version: string, message = "") {
+
+      var callingMethod = "";
+
+      try {
+        // Getting the name of the calling method through the stack trace.
+        callingMethod = (<any> new Error).stack
+          .split("\n")
+          .filter((step: string) => step.match(/http/))[1] // Just method names
```

Comment:
Unless there is a determined way to get the method name that isn't hacky (do a bit of research on this), we should probably go with a dumber approach (sending the methodName through to the function)